release_notes/fetch.py

```diff
@@ -417,14 +355,79 @@ def fetch_release_notes(
         github_api_lookup=github_api_lookup,
     )
 
-    release_notes: set[rnm.ReleaseNote] = {
-        rnm.create_release_notes_obj(
-            component_descriptor_lookup=component_descriptor_lookup,
-            version_lookup=version_lookup,
-            source_block=source_block,
-            source_component=component,
-            current_component=component,
-        ) for source_block in release_note_blocks
-    }
-
-    return release_notes
+    release_notes = [
+        release_note_block.as_release_note_entry(hostname=hostname)
+        for release_note_block in release_note_blocks
+    ]
+
+    if not release_notes:
+        return None
+
+    return rnm.ReleaseNotesDoc(
+        ocm=rnm.ReleaseNotesOcmRef(
+            component_name=component.name,
+            component_version=component.version,
+        ),
+        release_notes=release_notes,
+    )
+
+
+def collect_release_notes(
+    git_helper: gitutil.GitHelper,
+    release_version: str,
+    component: ocm.Component,
+    version_lookup: ocm.VersionLookup,
+    github_api_lookup: rnu.GithubApiLookup,
+    version_whither_ref_commit: git.Commit | None=None,
+    is_draft: bool=False,
+) -> tuple[rnm.ReleaseNotesDoc | None, list[rnm.ReleaseNotesDoc]]:
+    repo_path = git_helper.repo_path
+    local_release_notes_path = os.path.join(repo_path, '.ocm/release-notes')
+
+    release_notes_docs = list(rno.read_release_notes_from_dir(local_release_notes_path))
+    for release_notes_doc in release_notes_docs:
+        if (
+            not release_notes_doc.ocm
+            or not release_notes_doc.ocm.component_name
+            or not release_notes_doc.ocm.component_version
+        ):
+            release_notes_doc.ocm = rnm.ReleaseNotesOcmRef(
+                component_name=component.name,
+                component_version=component.version,
+            )
+
+    if fetched_release_notes_doc := fetch_release_notes(
+        component=component,
+        version_lookup=version_lookup,
+        git_helper=git_helper,
+        github_api_lookup=github_api_lookup,
+        version_whither=release_version,
+        version_whither_ref_commit=version_whither_ref_commit,
+        is_draft=is_draft,
+    ):
+        release_notes_docs.append(fetched_release_notes_doc)
+
+    grouped_release_notes_docs = rno.group_release_notes_docs(release_notes_docs)
+
+    component_id = component.identity()
+
+    component_matches = [
```

Comment:
naming-nit: is is more expressive to e.g. name this matching_release_notes (xxx_matches will rather imply a boolean value - or a list of matching components)